modules/c++/tiff/source/Common.cpp

```diff
@@ -54,7 +54,8 @@ sys::Uint64_T tiff::combine(sys::Uint32_T numerator,
 void tiff::split(sys::Uint64_T value, sys::Uint32_T &numerator,
                  sys::Uint32_T &denominator)
 {
-    numerator = ((sys::Uint32_T *)&value)[0];
-    denominator = ((sys::Uint32_T *)&value)[1];
+    const sys::Uint32_T* pValue = reinterpret_cast<sys::Uint32_T*>(&value);
+    numerator = pValue[0];
+    denominator = pValue[1];
```

Comment:
Casting sys::Uint64_T* to sys::Uint32_T* and dereferencing the result violates the strict aliasing rule and is undefined behavior. It tends to work for some compiler/option combinations (e.g. gcc with -fno-strict-aliasing), but it is not standard.

A standard compliant way to do this (until we can use C++20 bit_cast) is
```
const auto* buf = reinterpret_cast<sys::ubyte*>(&value); // reinterpret casting to char/unsigned char is allowed by the standard
memcpy(&numerator, buf, sizeof(sys::Uint32_T));
memcpy(&denominator, buf + sizeof(sys::Uint32_T), sizeof(sys::Uint32_T));
```

These give some details on the aliasing rules in C++. There are probably many other places in coda-oss with similar issues.
https://gist.github.com/shafik/848ae25ee209f698763cffee272a58f8
https://en.cppreference.com/w/cpp/language/reinterpret_cast#Type_aliasing

Reply:
Good call! I submitted an update